docs/compilation/performance_tips.md

```diff
@@ -0,0 +1,505 @@
+# Performance tips
+
+The most costly operation in Concrete is the table lookup operation, so the primary goal of optimizing performance is to reduce the amount of table lookups and reduce the bit width of the inputs to table lookups.
+
+## Try different implementation strategies for complex operations
+
+Concrete provides multiple implementation strategies for these complex operations:
+
+- [comparisons (<,<=,==,!=,>=,>)](../core-features/comparisons.md)
+- [bitwise operations (<<,&,|,^,>>)](../core-features/bitwise.md)
+- [minimum and maximum operations](../core-features/minmax.md)
+- [multivariate extension](../core-features/extensions.md#fhemultivariatefunction)
+
+{% hint style="info" %}
+The default strategy is the one that doesn't increase the input bit width, even if it's less optimal than the others. If you don't care about the input bit widths (e.g., if the inputs are only used in this operation), you should definitely change the default strategy.
+{% endhint %}
+
+Choosing the correct strategy can lead to big speedups. So if you are not sure which one to use, you can compile with different strategies and compare the complexity.
+
+For example, the following code:
+
+```python
+import numpy as np
+from concrete import fhe
+
+def f(x, y):
+    return x & y
+
+inputset = fhe.inputset(fhe.uint3, fhe.uint4)
+strategies = [
+    fhe.BitwiseStrategy.ONE_TLU_PROMOTED,
+    fhe.BitwiseStrategy.THREE_TLU_CASTED,
+    fhe.BitwiseStrategy.TWO_TLU_BIGGER_PROMOTED_SMALLER_CASTED,
+    fhe.BitwiseStrategy.TWO_TLU_BIGGER_CASTED_SMALLER_PROMOTED,
+    fhe.BitwiseStrategy.CHUNKED,
+]
+
+for strategy in strategies:
+    compiler = fhe.Compiler(f, {"x": "encrypted", "y": "encrypted"})
+    circuit = compiler.compile(inputset, bitwise_strategy_preference=strategy)
+    print(
+        f"{strategy:>55} "
+        f"-> {circuit.programmable_bootstrap_count:>2} TLUs "
+        f"-> {int(circuit.complexity):>12_} complexity"
+    )
+```
+
+prints:
+
```

Comment:
Could be great to find a example where the chunked one is not the better one

Reply:
Done, I've also showed comparison in the process.